components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.server/src/main/java/org/wso2/carbon/identity/configuration/mgt/server/cors/internal/impl/CORSServiceImpl.java

```diff
@@ -0,0 +1,288 @@
+                    }
+                    throw new CORSServiceException(String.format("Tenant %s already has %s as a CORS URL.",
+                            tenantDomain, url));
+                }
+            }
+
+            for (String url : urls) {
+                Attribute attribute = new CORSUrlToAttribute().apply(url);
+                getConfigurationManager().addAttribute(CORS_URL_RESOURCE_TYPE, CORS_URL_RESOURCE_NAME, attribute);
+            }
+        } catch (ConfigurationManagementException e) {
+            throw new CORSServiceException("Error while adding CORS URL.", e);
+        } finally {
+            endTenantFlow();
+        }
+    }
+
+    @Override
+    public void deleteCORSUrls(String tenantDomain, List<String> urls) throws CORSServiceException {
+
+        try {
+            startTenantFlow(tenantDomain);
+
+            for (String url : urls) {
+                if (isInvalidUrl(url)) {
+                    if (log.isErrorEnabled()) {
+                        log.error(String.format("%s is an invalid URL.", url));
+                    }
+                    throw new CORSServiceException(String.format("%s is an invalid URL.", url));
+                }
+                if (!tenantHasCORSUrl(url)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format(String.format("Tenant %s doesn't have %s as a CORS URL.",
+                                tenantDomain, url)));
+                    }
+                    throw new CORSServiceException(String.format("Tenant %s doesn't have %s as a CORS URL.",
+                            tenantDomain, url));
+                }
+            }
+
+            for (String url : urls) {
+                Attribute attribute = new CORSUrlToAttribute().apply(url);
+                getConfigurationManager().deleteAttribute(CORS_URL_RESOURCE_TYPE, CORS_URL_RESOURCE_NAME,
+                        attribute.getKey());
+            }
+        } catch (ConfigurationManagementException e) {
+            throw new CORSServiceException("Error while deleting CORS URLs.", e);
+        } finally {
+            endTenantFlow();
+        }
+    }
+
+    /**
+     * Setting the tenant for the scenarios where the tenant is unavailable in context.
+     *
+     * @param tenantDomain The tenant domain.
+     * @return void
+     */
+    private void startTenantFlow(String tenantDomain) {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Tenant flow started for %s.", tenantDomain));
+        }
+    }
+
+    /**
+     * End the tenant flow started in startTenantFlow.
+     *
+     * @return void
```

Comment:
For void methods, we do not have to specify the return type. Shall we remove it and fix other related places?